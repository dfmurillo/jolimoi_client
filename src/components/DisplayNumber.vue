<template>
  <div class="row">
    <div
      class="jumbotron col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto form p-5"
    >
      <div class="container">
        <p class="lead" v-text="title"></p>
        <h1 class="display-4 text-center" :data-test="id" v-text="number"></h1>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    id: {
      type: [Number, String],
      required: true
    },
    title: {
      type: String,
      required: true
    },
    number: {
      type: [Number, String],
      required: true
    }
  }
};
</script>
