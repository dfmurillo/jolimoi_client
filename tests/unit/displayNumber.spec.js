import { shallowMount } from "@vue/test-utils";
import DisplayNumber from "@/components/DisplayNumber.vue";

describe("DisplayNumber.vue", () => {
  let component;

  beforeEach(() => {
    component = shallowMount(DisplayNumber, {
      propsData: {
        title: "Test Title for component",
        number: 5,
        id: "MyId"
      }
    });
  });

  it("has the expected html structure", () => {
    expect(component.element).toMatchSnapshot();
  });
});
