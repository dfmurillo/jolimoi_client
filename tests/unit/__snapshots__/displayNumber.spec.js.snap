// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayNumber.vue has the expected html structure 1`] = `
<div
  class="row"
>
  <div
    class="jumbotron col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto form p-5"
  >
    <div
      class="container"
    >
      <p
        class="lead"
      >
        Test Title for component
      </p>
       
      <h1
        class="display-4 text-center"
        data-test="MyId"
      >
        5
      </h1>
    </div>
  </div>
</div>
`;
